nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/api/ElasticSearchAdmin.java

```diff
@@ -233,4 +234,10 @@ default void dropAndInitRepositoryIndex(String repositoryName) {
      */
     boolean useExternalVersion();
 
+    /**
+     * Returns the hint by the Elasticsearch operator name.
+     *
+     * @since 11.1
+     */
+    Optional<ESHintQueryBuilder> getHintByOperator(String name);
```

Comment:
What about having a default impl returning `Optional.empty()`?

Reply:
yes i see, if i did it, it means the call [see](https://github.com/nuxeo/nuxeo/blob/9e777e64730d788ffd2c8ce17028de4f85c43e94/nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/query/NxqlQueryConverter.java#L411) will returns a `null` and we will end up by a NPE error.

As i mentioned, this backport from master will be 100 % safe if there no customer / client that inherited from theses classes otherwise i don't see how we can make the nested operator without the contribution :)

Reply:
> yes i see, if i did it, it means the call see will returns a null and we will end up by a NPE error.
Having a default implementation returning `Optional.empty()` will throw a `UnsupportedOperationException`, it won't return `null` 🤔 
Anyway, it's probably better to have a default impl to new interface methods for a backport ...

> I think, this backport from master will be 100 % safe if there no customer / client that inherited from theses classes otherwise i don't see how we can make the nested operator without the contribution :)
We don't know that, so it's not 100% safe.

Reply:
yes, I prefer to have the exact backport in 10.10 (which was the case) even with the same typo :/

Reply:
_Having a default implementation returning Optional.empty() will throw a UnsupportedOperationException, it won't return null 🤔_:  sorry my comment wasn't clear enough, what i went to say it's the in the call [see](https://github.com/nuxeo/nuxeo/blob/9e777e64730d788ffd2c8ce17028de4f85c43e94/nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/query/NxqlQueryConverter.java#L411) i should return a ESBuilder if the optional returns empty we will end up with the `UnsupportedOperationException`. yes a default implementation will be better :)

_We don't know that, so it's not 100% safe_ i agree :)